pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1181,11 +1212,29 @@ func (og *operationGenerator) GenerateUnmapDeviceFunc(
 	actualStateOfWorld ActualStateOfWorldMounterUpdater,
 	mounter mount.Interface) (volumetypes.GeneratedOperations, error) {
 
-	blockVolumePlugin, err :=
-		og.volumePluginMgr.FindMapperPluginByName(deviceToDetach.PluginName)
-	if err != nil {
-		return volumetypes.GeneratedOperations{}, deviceToDetach.GenerateErrorDetailed("UnmapDevice.FindMapperPluginByName failed", err)
+	var blockVolumePlugin volume.BlockVolumePlugin
+	var err error
+	// Translate to CSI spec if migration enabled
+	if useCSIPlugin(og.volumePluginMgr, deviceToDetach.VolumeSpec) {
```

Comment:
I think we need to check for VolumeSpec == nil here, similar to GenerateUnmountVolumeFunc

Reply:
Didn't notice https://github.com/kubernetes/kubernetes/pull/74493. Updated

Reply:
Oh right it should be added to UnmapVolume.  When reconstruction fails (and spec == nil), then it only calls UnmountVolume => UnmapVolume.

Reply:
Updated